File: src/main/java/org/jukeboxmc/inventory/CraftingTableInventory.java
package org.jukeboxmc.inventory;

import com.nukkitx.protocol.bedrock.data.inventory.ContainerType;
import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.item.Item;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class CraftingTableInventory extends ContainerInventory {

    public CraftingTableInventory( InventoryHolder holder ) {
        super( holder, -1, 9 );
    }

    @Override
    public Player getInventoryHolder() {
        return (Player) this.holder;
    }

    @Override
    public @NotNull InventoryType getType() {
        return InventoryType.WORKBENCH;
    }

    @Override
    public @NotNull ContainerType getWindowTypeId() {
        return ContainerType.WORKBENCH;
    }

    @Override
    public void onOpen(@NotNull Player player ) {
        super.onOpen( player );
        player.setCraftingGridInventory( new BigCraftingGridInventory( player ) );
    }

    @Override
    public void onClose(@NotNull Player player ) {
        player.setCraftingGridInventory( new SmallCraftingGridInventory( player ) );
    }

    @Override
    public void setItem(int slot, @NotNull Item item, boolean sendContent ) {
        super.setItem( slot - 32, item, sendContent );
    }

    @Override
    public Item getItem( int slot ) {
        return super.getItem( slot - 32 );
    }
}

File: src/main/java/org/jukeboxmc/inventory/CraftingTableInventory.kt
package org.jukeboxmc.inventory

import com.nukkitx.protocol.bedrock.data.inventory.ContainerType
import org.jukeboxmc.item.Item
import org.jukeboxmc.player.Player

/**
 * @author LucGamesYT
 * @version 1.0
 */
class CraftingTableInventory(holder: InventoryHolder?) : ContainerInventory(holder, -1, 9) {
    override val inventoryHolder: InventoryHolder?
        get() = holder as Player
    override val type: InventoryType
        get() = InventoryType.WORKBENCH
    override val windowTypeId: ContainerType
        get() = ContainerType.WORKBENCH

    override fun onOpen(player: Player) {
        super.onOpen(player)
        player.craftingGridInventory = BigCraftingGridInventory(player)
    }

    override fun onClose(player: Player) {
        player.craftingGridInventory = SmallCraftingGridInventory(player)
    }

    override fun setItem(slot: Int, item: Item, sendContent: Boolean) {
        super.setItem(slot - 32, item, sendContent)
    }

    override fun getItem(slot: Int): Item? {
        return super.getItem(slot - 32)
    }
}